modules/afsocket/afunix-source.c

```diff
@@ -85,6 +86,14 @@ static gboolean
 afunix_sd_init(LogPipe *s)
 {
   AFUnixSourceDriver *self = (AFUnixSourceDriver *) s;
+ 
+  if (!is_file_socket(self->filename))
+    {
+      msg_warning("Source is not a unix socket",
+                evt_tag_str("filename", self->filename),
+                NULL);
+      return FALSE;
+    }
 
```

Comment:
File socket checking before `afsocket_sd_init_method` is must because `afsocket_sd_init_method` will be false if file is not a socket and syslog-ng is writting only the `Address already in use ...` error message.